R/zchunk_L244.building_USA.R

```diff
@@ -0,0 +1,910 @@
+    A44.demandFn_serv <- get_data(all_data, "gcam-usa/A44.demandFn_serv")
+    A44.gcam_consumer <- get_data(all_data, "gcam-usa/A44.gcam_consumer")
+    A44.satiation_flsp <- get_data(all_data, "gcam-usa/A44.satiation_flsp")
+    A44.sector <- get_data(all_data, "gcam-usa/A44.sector")
+    A44.subsector_interp <- get_data(all_data, "gcam-usa/A44.subsector_interp")
+    A44.subsector_logit <- get_data(all_data, "gcam-usa/A44.subsector_logit")
+    A44.subsector_shrwt <- get_data(all_data, "gcam-usa/A44.subsector_shrwt")
+    A44.globaltech_cost <- get_data(all_data, "gcam-usa/A44.globaltech_cost")
+    A44.globaltech_eff <- get_data(all_data, "gcam-usa/A44.globaltech_eff") %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year))
+    A44.globaltech_eff_avg <- get_data(all_data, "gcam-usa/A44.globaltech_eff_avg")
+    A44.globaltech_shares <- get_data(all_data, "gcam-usa/A44.globaltech_shares")
+    A44.globaltech_intgains <- get_data(all_data, "gcam-usa/A44.globaltech_intgains")
+    A44.globaltech_retirement <- get_data(all_data, "gcam-usa/A44.globaltech_retirement")
+    A44.globaltech_shrwt <- get_data(all_data, "gcam-usa/A44.globaltech_shrwt")
+    A44.globaltech_interp <- get_data(all_data, "gcam-usa/A44.globaltech_interp")
+    A44.demand_satiation_mult <- get_data(all_data, "gcam-usa/A44.demand_satiation_mult")
+    L144.flsp_bm2_state_res <- get_data(all_data, "L144.flsp_bm2_state_res")
+    L144.flsp_bm2_state_comm <- get_data(all_data, "L144.flsp_bm2_state_comm")
+    L144.in_EJ_state_comm_F_U_Y <- get_data(all_data, "L144.in_EJ_state_comm_F_U_Y")
+    L144.in_EJ_state_res_F_U_Y <- get_data(all_data, "L144.in_EJ_state_res_F_U_Y")
+    L143.HDDCDD_scen_state <- get_data(all_data, "temp-data-inject/L143.HDDCDD_scen_state") %>%
+      gather(year, value, starts_with("X")) %>%
+      mutate(year = as.integer(substr(year, 2, 5)))
+    L100.Pop_thous_state <- get_data(all_data, "L100.Pop_thous_state")
+    L100.pcGDP_thous90usd_state <- get_data(all_data, "L100.pcGDP_thous90usd_state")
+    # ===================================================
+    # Need to delete the buildings sector in the USA region (gcam.consumers and supplysectors)
+    L244.DeleteConsumer_USAbld <- tibble(region = "USA", gcam.consumer = A44.gcam_consumer_en$gcam.consumer)
+    L244.DeleteSupplysector_USAbld <- tibble(region = "USA", supplysector = A44.sector_en$supplysector)
+
+    # L244.SubregionalShares_gcamusa: subregional population and income shares (not currently used)
+    L244.SubregionalShares_gcamusa <- write_to_all_states(A44.gcam_consumer, c("region", "gcam.consumer")) %>%
+      mutate(pop.year.fillout = min(BASE_YEARS),
+             inc.year.fillout = min(BASE_YEARS),
+             subregional.population.share = 1,
+             subregional.income.share = 1)
+
+    # L244.PriceExp_IntGains_gcamusa: price exponent on floorspace and naming of internal gains trial markets
```

Comment:
I added this for demand. Couldn't find the function for floorspace. Do you know the file where it's located?

Reply:
Are waiting on the answer to this question to merge?

Reply:
@russellhz : I would check with Sha. I don't know the equation off the top of my head.

Reply:
<img width="575" alt="floorspace_function" src="https://user-images.githubusercontent.com/20807787/32762475-8249f8a8-c8c7-11e7-9342-10915646f9a8.png">
This is excerpted from Jiyong's core model proposal.

Reply:
In that equation, a stands for the satiation-adder, and unlike the comment in the text, it is never set to zero. Instead it is derived in the calibration years from an exogenous assumption of μ (all other parameters are known or otherwise assumed).

Reply:
Thanks Russel and Page!